msupply-foundation/open-msupply
# Add support to build Android with ndk 26 and Rust 1.76

The remaining issue was that ring 0.16.20 was not compiling correctly. To make it work some env variables needed to be pointed to the correct AR and CC binaries.



Fixes #2883

The remaining issue was that ring 0.16.20 was not compiling correctly. To
make it work some env variables needed to be pointed to the correct AR
and CC binaries.

Note, this might break the compilation using the older Rust and ndk versions, but I think if this patch is working for everybody we should just require the latest versions.

## Testing
- Install Android NDK 26.1.10909125
- Update to latest Rust (`rustup default stable`)
- Delete the `client/packages/android/server-lib` directory 
- Update your NDK_BIN env variable to the correct (check it points to the new ndk 26: `echo $NDK_BIN`)
- Compile the apk `yarn android:build:debug` (from client directory)
- Install the debug apk and test if it is running
